codalab/model/worker_model.py

```diff
@@ -325,7 +325,7 @@ def send_json_message(self, socket_id, message, timeout_secs, autoretry=True):
 
                 if not success:
                     # Shouldn't be too expensive just to keep retrying.
-                    time.sleep(0.003)
+                    time.sleep(0.3) # changed from 0.003 to keep from rate-limiting due to dead workers
```

Comment:
In principle, yes, expo backoff makes sense. Perhaps it will be more appropriate to switch all requests of this type to retry with expo back off in a separate PR tho.